### PR TITLE
Model deployment

### PR DESCRIPTION
**Pull request by oburakcinar**

### CODE REVIEW

File: src/main/java/io/passport/server/service/ModelDeploymentService.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+
+    /**
+     * Save a model deployment
+     * @param modelDeployment model deployment to be saved
+     * @return
+     */
+    public ModelDeployment saveModelDeployment(ModelDeployment modelDeployment) {
+        // Set the creation and last update timestamps
+        LocalDateTime now = LocalDateTime.now();
```
Comment (Okanmercan99) [inline src/main/java/io/passport/server/service/ModelDeploymentService.java:64]: Let's use Instant class for time class. Please update all time class usages accordingly.

File: src/main/java/io/passport/server/service/ModelDeploymentService.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    public Optional<ModelDeployment> updateModelDeployment(Long deploymentId, ModelDeployment updatedModelDeployment) {
+        Optional<ModelDeployment> oldModelDeployment = modelDeploymentRepository.findById(deploymentId);
+        if (oldModelDeployment.isPresent()) {
+            ModelDeployment modelDeployment = oldModelDeployment.get();
+
+            LocalDateTime now = LocalDateTime.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+            modelDeployment.setLastUpdatedAt(Timestamp.valueOf(now.format(formatter)));
+
+            Optional.ofNullable(updatedModelDeployment.getModelId()).ifPresent(modelDeployment::setModelId);
```
Comment (Okanmercan99) [inline src/main/java/io/passport/server/service/ModelDeploymentService.java:88]: Let's simplfy there as other service classes. We do not need silent null checking like this.

**Review by Okanmercan99 — CHANGES_REQUESTED** [review]

File: src/main/java/io/passport/server/model/ModelDeployment.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private String tags;
+
+    @Column(name = "identified_failures")
+    private String identifiedFailures;
+
+    @Column(name = "status")
+    private String status;
+
+    @Column(name = "created_at")
+    private Timestamp createdAt;
```
Comment (Okanmercan99) [inline src/main/java/io/passport/server/model/ModelDeployment.java:39]: Let's use instant objects here and all similiar usages also.

File: src/main/java/io/passport/server/service/ModelDeploymentService.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+     * @return
+     */
+    public Optional<ModelDeployment> updateModelDeployment(Long deploymentId, ModelDeployment updatedModelDeployment) {
+        Optional<ModelDeployment> oldModelDeployment = modelDeploymentRepository.findById(deploymentId);
+        if (oldModelDeployment.isPresent()) {
+            ModelDeployment modelDeployment = oldModelDeployment.get();
+
+            // set last updated timestamp
+            Instant now = Instant.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").withZone(ZoneId.systemDefault());
```
Comment (Okanmercan99) [inline src/main/java/io/passport/server/service/ModelDeploymentService.java:88]: You do not need to format time. You can pass Instant object directly to database.

**Review by Okanmercan99 — CHANGES_REQUESTED** [review]

**Review by Okanmercan99 — APPROVED** [review]

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -7,7 +7,7 @@ keycloak.ssl-required=external
 #keycloak client that is used for procedures
 keycloak.resource=AI4HF-Auth
 #Chosen keycloak client secret. Should be edited after each Docker compose for the Keycloak to work as intended
-keycloak.credentials.secret=pZlLseohp9n7dOiIwVkHNzswGCeYUH6O
+keycloak.credentials.secret=i9Gm2CbhusqLUGmamK25yvo9ZHf69QIj
```
Comment (dogukan10) [inline src/main/resources/application.properties:10]: Please revert this change back. You can change it in your local while developing the code.

**Pull request merged**